### PR TITLE
Improve quota

### LINKED ISSUE #10308: Add quota indicator for routing in the Builder UI
It is required to add some indicator of quota for the routing service in the builder, pretty similar to what was done in #10193 

Routing stuff is already in the user decorator. E.g: (from the js console)
```js
> user_data.mapzen_routing
Object {quota: 1500000, block_price: null, monthly_use: 828, hard_limit: true}
```

so I guess something like
```js
var routing = userModel.get('mapzen_routing');
```

can be used.

### LINKED ISSUE #10323: Improve feedback when not having quota
Ther are two scenarios that we are not controlling now.
a. User has not quota assigned. We should be more direct and say something like: "To get access to the Data Observatory/Routing/whatever function, [contact our team](link to contact)"

b. User has consumed all quota assigned, but has $0/1000 price for extra credits: "You have consumed all your credits during this billing cycle. Contact us to get some more".

Right now, the error is quite confusing :)

slack-imgs com

### PR DESCRIPTION
**Pull request by nobuti**
This PR fixes #10308 and fixes #10323.

screen shot 2016-10-27 at 18 22 23

screen shot 2016-10-27 at 18 23 21

cc @xavijam @saleiva

### CODE REVIEW

File: lib/assets/javascripts/cartodb3/editor/layers/layer-content-views/analyses/analyses-quota-factory.js [hunk 1]
```diff
@@ -39,56 +39,79 @@ function getProviderTypeByUser (type, userModel) {
 }
 
 var ANALYSES_WITH_QUOTA_MAP = {
-  'trade-area': function (userModel) {
-    var service = 'isolines_provider';
-    var provider = getProviderTypeByUser(service, userModel);
-    var isolines = userModel.get('here_isolines');
+  'trade-area': {
+    name: 'Trade area',
```
Comment (xavijam) [inline lib/assets/javascripts/cartodb3/editor/layers/layer-content-views/analyses/analyses-quota-factory.js]: Should we translate the names?

File: lib/assets/javascripts/cartodb3/editor/layers/layer-content-views/analyses/analyses-quota-factory.js [hunk 2]
```diff
@@ -39,56 +39,79 @@ function getProviderTypeByUser (type, userModel) {
...
+      return parseQuota(obsGeneral);
+    }
   },
-  'data-observatory-measure': function (userModel) {
-    var obsGeneral = userModel.get('obs_general');
-    return parseQuota(obsGeneral);
+  routing: {
+    name: 'Routing',
+    run: function (userModel) {
+      var routing = userModel.get('mapzen_routing');
```
Comment (xavijam) [inline lib/assets/javascripts/cartodb3/editor/layers/layer-content-views/analyses/analyses-quota-factory.js]: @CartoDB/data-services-admin-team do we only have `mapzen_routing` variable for routing?

**Comment by xavijam** [comment]
Good job man! I don't know if we should add . at the end of those messages appearing in the screenshots.

**Comment by xavijam** [comment]
Hey @CartoDB/dataservices, which are the routing analyses? These ones?

```
routing-sequential
routing-to-layer-all-to-all
routing-to-single-point
```

Which of them need quota checker?

**Comment by ethervoid** [comment]
AFAIK all of them

**Comment by xavijam** [comment]
Great, thanks @ethervoid. Which is the best staging server to test these UI changes and where the routing analyses are available?

**Comment by ethervoid** [comment]
Use ded04 :)

**Comment by nobuti** [comment]
@xavijam @javierarce @CartoDB/frontend for a future reference, we implemented the quota check for routing analyses in this PR but the thing is, the UI is not ready for these kind of analyses.

**Pull request merged**